java/client/src/main/java/glide/connectors/handlers/PubSubMessageQueue.java

```diff
@@ -0,0 +1,79 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.connectors.handlers;
+
+import glide.api.models.PubSubMessage;
+import java.util.LinkedList;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+/** A FIFO message queue for {@link PubSubMessage}. */
+public class PubSubMessageQueue {
+    // fields are protected to ease testing
+    /** The queue itself. */
+    protected final LinkedList<PubSubMessage> messageQueue = new LinkedList<>();
```

Comment:
LinkedLists tend to perform poorly, even for large data sets compared to ArrayList due to fitting better in the cache. See: https://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist-in-java

Reply:
I always add to the end and remove from the start. No reads performed.
Both operations take `O(1)`, so we can't make this faster.